### PR TITLE
Graphics kNN

### PR DESCRIPTION
**Pull request by juniorrojas**
Working visualization of kNN classification algorithm.

### CODE REVIEW

File: src/graphics/Graphics.hpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#ifndef __Graphics__
```
Comment (andreqi) [inline src/graphics/Graphics.hpp]: #pragma once

File: src/graphics/PixelBuffer.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
+#include "PixelBuffer.hpp"
+#include <SDL2/SDL.h>
+
+static const int BPP = 4;
```
Comment (andreqi) [inline src/graphics/PixelBuffer.cpp]: this is polluting the global environment, namespace maybe? or class static member?

File: src/graphics/Graphics.hpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+#include <SDL2/SDL.h>
+#include "PixelBuffer.hpp"
+#include <string>
+#include <vector>
+
+struct Label {
+  int index;
+  std::string name;
+  Uint32 color;
+  Uint32 lightColor;
```
Comment (andreqi) [inline src/graphics/Graphics.hpp]: http://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Variable_Names

File: src/graphics/Graphics.hpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+struct Sample {
+  std::vector<double> x;
+  Label* y;
+  Sample(std::vector<double> x, Label* y) : x(x), y(y) {}
+};
+
+class Graphics {
+private:
+  SDL_Window* window;
```
Comment (andreqi) [inline src/graphics/Graphics.hpp]: http://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Variable_Names

File: src/graphics/PixelBuffer.hpp [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#ifndef __PixelBuffer__
+#define __PixelBuffer__
+
+#include <SDL2/SDL.h>
+
+class PixelBuffer {
+private:
+  int width;
```
Comment (andreqi) [inline src/graphics/PixelBuffer.hpp]: name privates as name_

File: src/main.cc [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+int main(int argc, char** argv) {
+  Graphics graphics;
+
+  if (!graphics.init("Graphics")) return 1;
+
+  labels.push_back(new Label(0, "Red", 0xaa0000, 0xb04152));
+  labels.push_back(new Label(1, "Green", 0x0000aa, 0x4778ab));
+
+  samples.push_back(new Sample(std::vector<double>{50, 10}, labels[0]));
```
Comment (andreqi) [inline src/main.cc]: use list initialization

File: src/graphics/PixelBuffer.cpp (hunk 2 above)
Comment (juniorrojas) [inline src/graphics/PixelBuffer.cpp]: This and other constants will be added as class static members

**Pull request merged**